namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerInterface.scala

```diff
@@ -100,14 +100,16 @@ object ThriftNamerInterface {
     protected[this] def nextStamp(): Stamp
 
     protected[this] final def update(v: Try[T]): Unit = {
-      val obs = Observation(nextStamp(), v)
-      val promise = synchronized {
-        val previous = pending
-        current = Some(obs)
-        pending = new Promise[Observation]
-        previous
+      if (!current.exists(_.value == v)) {
+        val obs = Observation(nextStamp(), v)
+        val promise = synchronized {
+          val previous = pending
+          current = Some(obs)
```

Comment:
yeah, i think the real issue is that i _think_ it would have to be volatile if outside of the synchronized block to ensure visibility? i'm not a jmm lawyer...